azext_edge/edge/providers/orchestration/resources/brokers.py

```diff
@@ -85,6 +94,201 @@ def create(
             )
             return wait_for_terminal_state(poller, **kwargs)
 
+    def _build_tls_config(
+        self,
+        tls_auto_issuer_ref: Optional[str] = None,
+        tls_auto_duration: Optional[str] = None,
+        tls_auto_key_algo: Optional[str] = None,
+        tls_auto_key_rotation_policy: Optional[str] = None,
+        tls_auto_renew_before: Optional[str] = None,
+        tls_auto_san_dns: Optional[List[str]] = None,
+        tls_auto_san_ip: Optional[List[str]] = None,
+        tls_auto_secret_name: Optional[str] = None,
+        tls_manual_secret_ref: Optional[str] = None,
+    ):
+        config = {}
+        cm_config = defaultdict(dict)
+        man_config = defaultdict(dict)
+
+        if tls_auto_issuer_ref:
+            tls_auto_issuer_ref = parse_kvp_nargs(tls_auto_issuer_ref)
+            issuer_config = {}
+            for key in ["group", "kind", "name"]:
+                if key in tls_auto_issuer_ref:
+                    issuer_config[key] = tls_auto_issuer_ref[key]
+            if "group" not in issuer_config:
+                issuer_config["group"] = "cert-manager.io"
+            cm_config["issuerRef"] = issuer_config
+        if tls_auto_duration:
+            cm_config["duration"] = tls_auto_duration
+        if tls_auto_key_algo:
+            cm_config["privateKey"]["algorithm"] = tls_auto_key_algo
+        if tls_auto_key_rotation_policy:
+            cm_config["privateKey"]["rotationPolicy"] = tls_auto_key_rotation_policy
+        if tls_auto_renew_before:
+            cm_config["renewBefore"] = tls_auto_renew_before
+        if tls_auto_san_dns:
+            cm_config["san"]["dns"] = tls_auto_san_dns
+        if tls_auto_san_ip:
+            cm_config["san"]["ip"] = tls_auto_san_ip
+        if tls_auto_secret_name:
+            cm_config["secretName"] = tls_auto_secret_name
+
+        if tls_manual_secret_ref:
+            man_config["secretRef"] = tls_manual_secret_ref
+
+        if all([cm_config, man_config]):
```

Comment:
why not have this check happen earlier in line 111 so it fails early?

Reply:
The current placement is simpler (more readable) and ensures the check is performed after all potential configs have been gathered. It handles cases where different tls_auto_* keys are used without additional checks/conditions. The performance difference between the approaches is negligible as building the dictionaries in this scenario and comparing truthiness is negligible impact to performance.

Reply:
Oh I see! Coming from a UI lens, user don't even see the input because they choose either manual or auto before they can proceed. But because of the nature of CLI that all the input has been provided, you need to compare not a single value contrast